phoenix-spark/src/main/java/org/apache/phoenix/spark/datasource/v2/reader/PhoenixInputPartitionReader.java

```diff
@@ -77,14 +80,15 @@ private QueryPlan getQueryPlan() throws SQLException {
         String scn = options.getScn();
         String tenantId = options.getTenantId();
         String zkUrl = options.getZkUrl();
-        Properties overridingProps = new Properties();
+        Properties overridingProps = options.getOverriddenProps();
```

Comment:
I think options.getOverriddenProps can never be null here, right?

Reply:
Ideally they can never be null, but added a null-check to avoid any problems in the future. Same for the write path.